jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -689,10 +718,16 @@ def model_image(scidata_bkg, scierr, scimask, refmask, ref_files, box_weights, s
         tikfac_log_range = np.log10(tikfac) + np.array([-2, 8])
 
         # Model the remaining part of order 2 with atoca
-        model, spec_ord = model_single_order(scidata_bkg, scierr, ref_file_order,
-                                             mask_fit, global_mask, order,
-                                             pixel_wave_grid, valid_cols, save_tiktests,
-                                             tikfac_log_range=tikfac_log_range)
+        try:
+            model, spec_ord = model_single_order(scidata_bkg, scierr, ref_file_order,
+                                                 mask_fit, global_mask, order,
+                                                 pixel_wave_grid, valid_cols, save_tiktests,
+                                                 tikfac_log_range=tikfac_log_range)
+
+        except MaskOverlapError:
+            log.error('Not enough unmasked pixels to model the remaining part of order 2. Model and spectrum will be NaN in that spectral region.')
```

Comment:
Is this potential error really confined to only occurring for order 2, or could it occur for any order? Should we be trapping for this in other orders too?

Reply:
Order 1 covers the whole wavelength range, and as far as I know Order 3 is currently not used.  I'm happy to change this to check the other orders anyway though.